### PR TITLE
Use the right math and metrics to measure used Memory 

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
This was just plain wrong 😳 

It's right now, plus there are tooltips for exact numbers:

screen shot 2017-10-16 at 7 00 01 pm

### CODE REVIEW

File: ui/app/helpers/format-bytes.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * bytes is larger than one.
+ */
+export function formatBytes([bytes]) {
+  let unitIndex = 0;
+  while (bytes >= 1024 && unitIndex < UNITS.length - 1) {
+    bytes /= 1024;
+    unitIndex++;
+  }
+
+  return `${Math.floor(bytes)} ${UNITS[unitIndex]}`;
```
Comment (meirish) [inline ui/app/helpers/format-bytes.js]: this helper is 😗👌

File: ui/app/models/allocation.js [hunk 2]
```diff
@@ -30,18 +30,19 @@ export default Model.extend({
...
-      }
-      return used / total;
+  memoryUsed: computed.readOnly('stats.ResourceUsage.MemoryStats.RSS'),
+  cpuUsed: computed('stats.ResourceUsage.CpuStats.TotalTicks', function() {
+    return Math.floor(this.get('stats.ResourceUsage.CpuStats.TotalTicks'));
+  }),
+
+  percentMemory: computed('taskGroup.reservedMemory', 'memoryUsed', function() {
+    const used = this.get('memoryUsed') / 1024 / 1024;
+    const total = this.get('taskGroup.reservedMemory');
```
Comment (meirish) [inline ui/app/models/allocation.js]: does the taskGroup only support specifying in one unit? or will you have to do conversions here too?

Comment (DingoEatingFuzz) [reply ui/app/models/allocation.js]: It's always in MiB. In fact it's unitless in the job file: https://github.com/hashicorp/nomad/blob/master/terraform/examples/spark/hdfs.nomad#L74

File: ui/app/helpers/format-bytes.js (hunk 1 above)
Comment (thrashr888) [reply ui/app/helpers/format-bytes.js]: ooh nice one. hope you don't end up with gigs of bytes in the ui.

**Pull request merged**